service/internal/cloudconfig/template/aws-cni.go

```diff
@@ -222,6 +222,15 @@ spec:
             ## Explicit interface naming
             - name: AWS_VPC_K8S_CNI_VETHPREFIX
               value: eni
+            {{- if eq .AWSCNIPrefix true }}
+            ## Deviation from original manifest - 6
+            ## If CNI prefix validation is enabled it will improve the speed of allocating IPs on a nitro based node (e.g. m5.2xlarge) 
+            ## By setting a annotation on the Cluster CR it can be enabled or disabled.
+            - name: ENABLE_PREFIX_DELEGATION
+              value: "true"
+            - name: "WARM_PREFIX_TARGET"
+              value: "1"
+            {{- end }}
```

Comment:
@paurosello when prefix is enabled should we remove the warm IP target a minimum IP target envs?

Reply:
No, because that is still used for M4 family machines which do not support the Prefix but just double checked documentation and:

```
When the Prefix IP mode is utilized, you cannot set the values for WARM_PREFIX_TARGET, WARM_IP_TARGET, or MINIMUM_IP_TARGET variables to zero. The settings for these parameters will be determined by use cases. If set, WARM_IP_TARGET and/or MINIMUM_IP_TARGET will take precedence over WARM_PREFIX_TARGET.
```

So maybe it should be an OR -> If prefix enabled we remove WARM_IP_TARGET and MINIMUM_IP_TARGET

Reply:
that's what i mean, if prefix is enabled we should omit them otherwise prefix will not be used

Reply:
```
If set, WARM_IP_TARGET and/or MINIMUM_IP_TARGET will take precedence over WARM_PREFIX_TARGET.
```

Reply:
I added this small line here to remove envs in case of CNI prefix https://github.com/giantswarm/aws-operator/pull/3200/commits/076a3be292f61af09236cd9bc2ef73a0f2e0b858

Reply:
lgtm if pau is happy